### DISCUSSION TITLE
different suite hierarchy shown in test ops and allure report for the same test.

### DISCUSSION THREAD

**Discussion by Lev-v** [allure-framework/allure2#2759] [TestOps]
I use allure-pytest 2.13.5 and allure testops 4.26.5 I didn't find something related to my problem in the release notes so I don't think it's a version issue.

I use allure decorators in my tests:
parent_suite, suite, sub_suite and title

e.g:
`@allure.parent_suite(“my_parent_suite”)`
`@allure.suite(“my_suite”)`
`@allure.sub_suite(“my_sub_suite”)`
`class MyTest:`
`    @allure.title(“my_title”)`
`    def test_example(self):`
`		assert True`

Problem:
when I run tests locally and make a local report via "allure serve" I have everything working correctly and in the report under "suites" section I can see the following hierarchy: “my_parent_suite”-“my_suite”-“my_sub_suite”-“my_title”.
But when I transfer the same result files to allure test ops, in launches->results section I can select "suites" view and it does not show the whole hierarchy, but only “my_suite”-“my_title”. As far as I Can see - "parent_suite" and "sub-suite" sections are not displayed.
 **How to make the full hierarchy visible in test ops so it was the same as in the local report?**

![allure test ops question](https://github.com/user-attachments/assets/34a61912-d6a1-4b2f-ab87-60289d8823b5)

The only info I found about suite|parent_suite in the testops docs was about the custom fields, but adding the custom field "parent_suite" in Administration and the same label to my test resulted only in the addition of "Parent Suite: my_parent_suite" string, inside the test on the right side, and the hierarchy stayed the same

**Comment by EugeneBatenev** [comment]
Hello @Lev-v 

Allure TestOps and Allure Report are separate products with different internal logic. You mentioned that to create the required hierarchy, you are using labels like:

@allure.parent_suite("my_parent_suite")
@allure.suite("my_suite")
@allure.sub_suite("my_sub_suite")
from the Allure Report documentation: https://allurereport.org/docs/gettingstarted-navigation/#suite-based-hierarchy which is right.

However, Allure TestOps does not process these labels. In TestOps, hierarchies are built based on custom fields.

To start, you need to define the desired custom fields globally at the instance level and then in the specific project. You can find more details here: https://docs.qameta.io/allure-testops/briefly/test-cases/custom-fields/

Once you’ve added the necessary custom fields to your project, you'll need to set up the correct hierarchy. Go to the project settings, add a new hierarchy tree (or edit an existing one), and configure it using the custom fields you've defined. More details about this process can be found here: https://docs.qameta.io/allure-testops/briefly/project/trees/

Next, you need to properly label your automated tests so that when result files are uploaded, the test cases are correctly organized within the hierarchy based on the metadata. You can learn more about this here: https://docs.qameta.io/allure-testops/briefly/test-cases/labels/

**Comment by cheshi-mantu** [comment]
@Lev-v BTW, there dedicated support portal https://help.qameta.io for questions related to Allure TestOps. Please register with the corporate email and we'll be happy to assist.

**Comment by Lev-v** [comment]
thx for the hints, everything works fine
